include/mbedtls/ssl.h

```diff
@@ -544,6 +555,9 @@ typedef enum
     MBEDTLS_SSL_HANDSHAKE_OVER,
     MBEDTLS_SSL_SERVER_NEW_SESSION_TICKET,
     MBEDTLS_SSL_SERVER_HELLO_VERIFY_REQUEST_SENT,
+#if defined(MBEDTLS_SSL_PROTO_TLS1_3_EXPERIMENTAL)
+    MBEDTLS_SSL_ENCRYPTED_EXTENSIONS,
+#endif /* MBEDTLS_SSL_PROTO_TLS1_3_EXPERIMENTAL */
```

Comment:
Should we have a separate field `tls13_state` for now? (Considering we don't negotiate "1.2 or 1.3" yet, but decide about it in the config.) That way, we can do things cleanly in the new 1.3 code (have an enum rather than `int`, choose clear names and document their meaning, etc.) without being burdened with the debt from 1.2?

And then we'll clean up 1.2 when we implement 1.2 vs 1.3 negotiation and see it it's better to merge the states or not, and how to improve the 1.2 states.

Wdyt?

Reply:
Hmm, I'm no longer so sure about this suggestion. Now I'm afraid this would add complexity in several places that check the state: for code paths that are 1.2-only this is not a problem, but code paths that are shared (at a minimum, `mbedtls_ssl_read()`, `mbedtls_ssl_write()`, `mbedtls_ssl_handshake()` but perhaps several others) would need to be adapted to check both `ssl->state` and `ssl->tls13_state` and combine the result correctly...

Reply:
How about next steps? 

My suggestion is keep current status in this PR.

Reply:
I prefer to keep the PR as is and change it later, e.g. when also improving the documentation of the 1.2 state machine.

Reply:
ok to not change that in this PR.